src/main/scala/com/typesafe/sbt/SbtProguard.scala

```diff
@@ -26,19 +32,30 @@ object SbtProguard extends AutoPlugin {
     proguardVersion := "7.0.0",
     proguardDirectory := crossTarget.value / "proguard",
     proguardConfiguration := proguardDirectory.value / "configuration.pro",
-    artifactPath := proguardDirectory.value / (artifactPath in packageBin in Compile).value.getName,
+    artifactPath := proguardDirectory.value / ( Compile / packageBin / artifactPath).value.getName,
     managedClasspath := Classpaths.managedJars(configuration.value, classpathTypes.value, update.value),
-    proguardBinaryDeps := getAllBinaryDeps.value,
-    proguardInputs := (fullClasspath in Runtime).value.files,
-    proguardLibraries := proguardBinaryDeps.value filterNot proguardInputs.value.toSet,
+    proguardBinaryDeps := {
+      (Compile / libraryDependencies).value match {
+        case analysis: Analysis =>
+          analysis.relations.allLibraryDeps.collect { case vf: PathBasedFile =>
+            vf.toPath.toFile
+          }.toSeq
+      }
+    },
+    proguardInputs := {
+      assembly.value
+      Seq((assembly / assemblyOutputPath).value)
+    },
+    (proguard / javaHome) := Some(FileSystems.getDefault.getPath(sys.env("JAVA_HOME")).toFile),
```

Comment:
Should we be depending on `JAVA_HOME` being set?

Reply:
Good question. This is the only way I could get it to work for me. Otherwise it couldn't find the Java standard library

Reply:
`System.getProperty("java.home")`?